src/mi/backtrace.c

```diff
@@ -69,12 +70,52 @@ unw_backtrace (void **buffer, int size)
   if (unlikely (tdep_trace (&cursor, buffer, &n) < 0))
     {
       unw_getcontext (&uc);
-      return slow_backtrace (buffer, size, &uc);
+      return slow_backtrace (buffer, size, &uc, 0);
     }
 
   return n;
 }
 
+int
+unw_backtrace2 (void **buffer, int size, unw_context_t* uc2)
```

Comment:
could we handle any `unw_context_t *` object by adding a `int flag` to the signature, similar to `unw_init_local2`? Use case: foo calls bar calls baz. bar calls unw_getcontext, passes the uc2 to baz, and baz calls unw_backtrace2 which returns the callstack from bar's context.

I think only change would be to signature and changing:
```
if (unlikely (unw_init_local2 (&cursor, &uc, flag) < 0))
```
In the case of signal handler, call this function with flag==UNW_INIT_SIGNAL_FRAME. In the case of no signal handler, call with flag==0

Reply:
Here is an example diff between e831ee22d7356d46f65ad7a93553f54e235d9c1e and the proposed change in the comment above:
```
diff --git a/src/mi/backtrace.c b/src/mi/backtrace.c
index 8fea3444..0f2863fc 100644
--- a/src/mi/backtrace.c
+++ b/src/mi/backtrace.c
@@ -77,7 +77,7 @@ unw_backtrace (void **buffer, int size)
 }
 
 int
-unw_backtrace2 (void **buffer, int size, unw_context_t* uc2)
+unw_backtrace2 (void **buffer, int size, unw_context_t* uc2, int flag)
 {
   if (size == 0)
     return 0;
@@ -89,7 +89,7 @@ unw_backtrace2 (void **buffer, int size, unw_context_t* uc2)
   // need to copy, because the context will be modified by tdep_trace
   unw_context_t uc = *(unw_context_t*)uc2;
 
-  if (unlikely (unw_init_local2 (&cursor, &uc, UNW_INIT_SIGNAL_FRAME) < 0))
+  if (unlikely (unw_init_local2 (&cursor, &uc, flag) < 0))
     return 0;
 
   // get the first ip from the context
@@ -110,7 +110,7 @@ unw_backtrace2 (void **buffer, int size, unw_context_t* uc2)
   // and add 1 to it (the one we retrieved above)
   if (unlikely (tdep_trace (&cursor, buffer, &n) < 0))
     {
-      return slow_backtrace (buffer, remaining_size, &uc, UNW_INIT_SIGNAL_FRAME) + 1;
+      return slow_backtrace (buffer, remaining_size, &uc, flag) + 1;
     }
 
   return n + 1;
```

Reply:
done